server/src/test/java/org/opensearch/search/SearchServiceTests.java

```diff
@@ -1413,6 +1414,116 @@ public void testConcurrentSegmentSearchSearchContext() throws IOException {
             .get();
     }
 
+    /**
+     * Tests that the slice count is calculated correctly when concurrent search is enabled
+     *  If concurrent search enabled -
+     *       pick index level slice count setting if index level setting is set
+     *       else pick default cluster level slice count setting
+     * @throws IOException
+     */
+    public void testConcurrentSegmentSearchSliceCount() throws IOException {
+        Integer[][] scenarios = {
+            // cluster setting, index setting, expected slice count
+            // expected value null will pick up default value from settings
+            { null, null, null },
```

Comment:
```suggestion
            { null, null, clusterService.getClusterSettings().get(SearchService.CONCURRENT_SEGMENT_SEARCH_TARGET_MAX_SLICE_COUNT_SETTING) },
```

Reply:
Can we can avoid L1463 this way?

Reply:
We can, but i'd have to move the scenarios below to a point after getting the `clusterService`